java/examples/src/java/org/apache/orc/examples/InMemoryCryptionWriter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc.examples;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.*;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class InMemoryCryptionWriter {
+    public static void main(Configuration conf, String[] args) throws IOException {
+        TypeDescription schema =
+                TypeDescription.fromString("struct<x:int,y:string>");
+        // the kmsKey used to encrypt data.
+        byte[] kmsKey = "secret123".getBytes(StandardCharsets.UTF_8);
+        // The primary use of InMemoryKeystore is for used who doesn't have a
+        // Hadoop KMS.
+        InMemoryKeystore keyProvider = new InMemoryKeystore()
+                .addKey("pii", EncryptionAlgorithm.AES_CTR_128, kmsKey);
+        // "encryption" used to specify columns and kmsKey which will be encrypted
+        String encryption = "pii:x,y";
+        Writer writer = OrcFile.createWriter(new Path("my-cryption.orc"),
```

Comment:
Maybe, shall we rename `my-cryption.orc` into `encrypted.orc`?

In the document, we use `encrypted` as a table name, https://orc.apache.org/docs/spark-config.html

Reply:
Sure, done.